### PR TITLE
Add Netscaler Support

### PR DESCRIPTION
**Pull request by tombo315**

### CODE REVIEW

**Comment by ktbyers** [comment]
@tombo315 Can you run the unit tests against this? See https://github.com/ktbyers/netmiko/blob/develop/TESTING.md

**Comment by ktbyers** [comment]
I will probably make these modifications specified here:

https://github.com/ktbyers/netmiko/commit/3ba8af66d0c6ebbdd4a3799b819a72292da2c6c5

**Comment by tombo315** [comment]
I'll have a look at the tests and attach some output.

**Comment by ktbyers** [comment]
Okay, sounds good. Let me know if you have any questions.

**Comment by tombo315** [comment]
```
$ pytest -v test_netmiko_show.py --test_device netscaler
================================================================================= test session starts =================================================================================
platform linux -- Python 3.6.3, pytest-3.5.0, py-1.5.3, pluggy-0.6.0 -- /home/thomasbo/py36/bin/python3.6
cachedir: ../.pytest_cache
rootdir: /home/thomasbo/netmiko, inifile:
collected 12 items                                                                                                                                                                    

test_netmiko_show.py::test_disable_paging PASSED                                                                                                                                [  8%]
test_netmiko_show.py::test_ssh_connect PASSED                                                                                                                                   [ 16%]
test_netmiko_show.py::test_ssh_connect_cm PASSED                                                                                                                                [ 25%]
test_netmiko_show.py::test_send_command_timing PASSED                                                                                                                           [ 33%]
test_netmiko_show.py::test_send_command_expect PASSED                                                                                                                           [ 41%]
test_netmiko_show.py::test_base_prompt PASSED                                                                                                                                   [ 50%]
test_netmiko_show.py::test_strip_prompt PASSED                                                                                                                                  [ 58%]
test_netmiko_show.py::test_strip_command PASSED                                                                                                                                 [ 66%]
test_netmiko_show.py::test_normalize_linefeeds PASSED                                                                                                                           [ 75%]
test_netmiko_show.py::test_clear_buffer PASSED                                                                                                                                  [ 83%]
test_netmiko_show.py::test_enable_mode PASSED                                                                                                                                   [ 91%]
test_netmiko_show.py::test_disconnect PASSED                                                                                                                                    [100%]

============================================================================= 12 passed in 27.90 seconds ==============================================================================```

**Comment by tombo315** [comment]
I don't have a device handy to test config changes, but these devices lack any describable "config mode". 

```
set <whatever>
```

File: tests/etc/responses.yml.example [hunk 1]
```diff
@@ -104,6 +104,15 @@ alcatel_aos:
   version_banner: "Alcatel-Lucent OS6250-24 6.7.1.108.R04 Service Release, January 04, 2017"
   multiple_line_output: "FC - ForcedCopper PC - PreferredCopper C - Copper"
 
+
+netscaler:
+  base_prompt: ""
```
Comment (tombo315) [inline tests/etc/responses.yml.example]: @ktbyers this required a test-change--not the ideal "fix", but I'm not sure it's applicable for these devices

**Comment by ktbyers** [comment]
@tombo315 I made a few other changes here (basically caused config_mode to be a no-op and then we didn't need the special send_config_set).

I also made self.base_prompt be '>' if that was all the prompt contained as I was concerned about other things breaking if this was a null string (so netscaler has a special `set_base_prompt` method).

It would be good if you could regression test these changes.

Does Netscaler have any notion of enable mode?

Also what does 'save ns config' look like on the CLI? I assume this is the standard way of saving the running config to startup config (at least that looked like the command online).

Basically, I want to add a `save_config` method, but I need to see the behavior of what the CLI looks like to do this.
